File: Manual/LightField/Add-in and Automation SDK/Samples/VisualBasic Add-Ins/Acquire Async Sample/AcquireAsyncSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Markup
Imports System.AddIn
Imports System.IO
Imports System.AddIn.Pipeline
Imports PrincetonInstruments.LightField.AddIns

''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
'''''''''' This sample is intended to show how to acquire data (to the default name and location) ''''''''''
'''''''''' and demonstrate how to hook into the acquisition started                               ''''''''''
'''''''''' and acquisition completed events.                                                      ''''''''''
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Public Class AddinAcquireAsyncSample
    Inherits AddInBase

    Public control_ As AsynAcquireSample

    Public ReadOnly Property UISupport As UISupport
        Get
            Return UISupport.ExperimentSetting
        End Get
    End Property

    Public Sub Activate(ByVal app As ILightFieldApplication)
        'Capture Interface
        LightFieldApplication = app
        'Build your controls
        control_ = New AsynAcquireSample(LightFieldApplication)
        ExperimentSettingElement = control_
    End Sub

    Public Sub Deactivate()
    End Sub

    Public Shadows ReadOnly Property UIExperimentSetting As Windows.FrameworkElement
        Get
            If ExperimentSettingElement Is Nothing Then
                Throw New NotImplementedException("UIExperiment has no implementation!")
            Else
                Return ExperimentSettingElement
            End If
        End Get
    End Property

    Public Shadows ReadOnly Property UIExperimentSettingTitle As String
        Get
            Return "Acquire Asynchronous Sample"
        End Get
    End Property

End Class

File: Manual/LightField/Add-in and Automation SDK/Samples/CSharp Add-Ins/Acquire Async Sample/AcquireAsyncSample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Markup;
using System.AddIn;
using System.IO;
using System.AddIn.Pipeline;

using PrincetonInstruments.LightField.AddIns;

using LightFieldAddInSamples;

namespace LightFieldAddInSamples
{   
    ///////////////////////////////////////////////////////////////////////////
    // This sample is intended to show how to acquire data(to the default name
    // and location) and demonstrate how to hook into the acquisition started
    // and acquisition completed events.
    ///////////////////////////////////////////////////////////////////////////
    [AddIn("Acquire Asynchronous Sample (C#)", 
            Version = "1.0.0",
            Publisher = "Teledyne Princeton Instruments", 
            Description = "Shows an example of asynchronous acquisition and how to use events")]
    [QualificationData("IsSample", "True")]
    public class AddinAcquireAsyncSample : AddInBase, ILightFieldAddIn
    {
        private Acquire_Async_Sample.AsynAcquireSample control_;

        ///////////////////////////////////////////////////////////////////////
        public UISupport UISupport { get { return UISupport.ExperimentSetting; } }
        ///////////////////////////////////////////////////////////////////////
        public void Activate(ILightFieldApplication app)
        {
            // Capture Interface
            LightFieldApplication = app;

            // Build your controls
            control_ = new Acquire_Async_Sample.AsynAcquireSample(LightFieldApplication);
            ExperimentSettingElement = control_;
        }
        ///////////////////////////////////////////////////////////////////////
        public void Deactivate() { }
        ///////////////////////////////////////////////////////////////////////        
        ///////////////////////////////////////////////////////////////////////
        public override string UIExperimentSettingTitle { get { return "Acquire Asynchronous Sample"; } }       
    }
}
